pkg/image/api/types.go

```diff
@@ -60,6 +65,23 @@ type ImageLayer struct {
 	Size int64
 }
 
+// ImageSignatureType represents a type of image's signature.
+type ImageSignatureType string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type ImageSignatureType
+	// An opaque binary string which is an image's signature.
+	Blob string
+	// Optional, if specified, it is the time of signature's creation. Should match a timestamp in the blob.
```

Comment:
"Should match a timestamp in the blob." is not compatible with the system auto-assigning a timestamp without any knowledge of the blob